osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs

```diff
@@ -196,6 +195,15 @@ private void scoreChanged(ValueChangedEvent<ScoreInfo?> score)
             if (!(hitEvents.CalculateMedianHitError() is double median))
                 return;
 
+            // affecting unstable rate here is used as a substitute of determining if a hit event represents a *timed* hit event,
+            // i.e. an user input that the user had to *time to the track*,
+            // i.e. one that it *makes sense to use* when doing anything with timing and offsets.
+            bool hasEnoughUsableEvents = hitEvents.Count(HitEventExtensions.AffectsUnstableRate) >= 50;
+
+            // If we are already displaying a score, continue displaying it rather than showing the user "play too short" message.
```

Comment:
I don't know what it is about this comment but it made things more difficult to understand. Would propose

```suggestion
            // If we already have an old score with enough hit events and the new score doesn't have enough, continue displaying the old one rather than showing the user "play too short" message.
```